lib/octicons_gem/lib/octicons_v2/octicon_v2.rb

```diff
@@ -84,5 +81,25 @@ def calculate_width(height)
     def calculate_height(width)
       (width.to_i * @height) / @width
     end
+
+    def get_octicon(symbol, options = {})
+      if octicon = OcticonsV2::OCTICON_SYMBOLS[symbol]
+        height = options[:height] || options[:width] || 16
+        octicon_height = closest_octicon_height(octicon["heights"].keys, height)
```

Comment:
Does every icon have a 24px version? If so, it could probably simplify this logic, which may run thousands of time on a single page 😬

Reply:
Right now, we can't assume that every icon has a 24px version. If it leads to a significant performance improvement, I'm open to enforcing that every icon has a 16px and 24px version. How would that simplify this logic?

Reply:
This is a little rough, but maybe something as simple as:

```suggestion
        octicon_height = height > 16 ? 32 : 16
```

Using constants of course 😉

Reply:
Hmm, I'm hesitant about requiring all icons to have a 16px and 24px version for a couple reasons:

- Logos don't really need 16px and 24px versions since we can scale the logo svgs to any size without them looking weird.  Requiring that each logo has a 16px and 24px version (e.g. `mark-github-16.svg` and `mark-github-24.svg`) that are identical seems cumbersome.

- Hard coding the sizes that each icon must have would also make adding smaller or larger versions of icons in the future much more difficult. For example, adding a smaller sized icon (say 12px) would require us design a complete 12px set. This would make it hard for us to ship things in an "incrementally correct" way.

Reply:
Is there a way we can measure the performance impact of the `closest_octicon_height` function?

Reply:
@colebemis those are fair points. I think you're right to push us towards benchmarking.

https://github.com/evanphx/benchmark-ips is what I use. I'd be happy to 🍐 on this ❤️